pkg/cloudprovider/backup_service.go

```diff
@@ -194,17 +203,30 @@ func (br *backupService) DeleteBackupDir(bucket, backupName string) error {
 	return errors.NewAggregate(errs)
 }
 
-func (br *backupService) CreateBackupSignedURL(backupType api.DownloadTargetKind, bucket, backupName string, ttl time.Duration) (string, error) {
-	switch backupType {
+func (br *backupService) CreateSignedURL(target api.DownloadTarget, bucket string, ttl time.Duration) (string, error) {
+	switch target.Kind {
 	case api.DownloadTargetKindBackupContents:
-		return br.objectStorage.CreateSignedURL(bucket, getBackupContentsKey(backupName), ttl)
+		return br.objectStorage.CreateSignedURL(bucket, getBackupContentsKey(target.Name), ttl)
 	case api.DownloadTargetKindBackupLog:
-		return br.objectStorage.CreateSignedURL(bucket, getBackupLogKey(backupName), ttl)
+		return br.objectStorage.CreateSignedURL(bucket, getBackupLogKey(target.Name), ttl)
+	case api.DownloadTargetKindRestoreLog:
+		// restore name is formatted as <backup name>-<timestamp>
+		backup := strings.Split(target.Name, "-")[0]
+		return br.objectStorage.CreateSignedURL(bucket, getRestoreLogKey(backup, target.Name), ttl)
 	default:
-		return "", fmt.Errorf("unsupported download target kind %q", backupType)
+		return "", fmt.Errorf("unsupported download target kind %q", target.Kind)
 	}
 }
 
+func (br *backupService) UploadRestoreLog(bucket, backup, restore string, log io.ReadSeeker) error {
+	key := getRestoreLogKey(backup, restore)
+	return br.objectStorage.PutObject(bucket, key, log)
+}
+
+func (br *backupService) CreateRestoreLogSignedURL(bucket, backupName, restoreName string, ttl time.Duration) (string, error) {
```

Comment:
is this used/needed?

Reply:
I deleted it from one spot but missed this one.